### ISSUE TITLE
Launching Edizon Menu Crashes  Atmosphere

### ISSUE THREAD

**Issue by ReZeroEmiria** [WerWolv/EdiZon#136]
![Error Screen](https://user-images.githubusercontent.com/31756219/79414234-17c7e200-7f78-11ea-8d25-d179a2d163fd.png)

I've used the following:
https://github.com/WerWolv/nx-ovlloader/releases/tag/v1.0.4
https://github.com/WerWolv/Tesla-Menu/releases/tag/v1.1.2
And the Edizon overlay after updating to 10.0.0 and Atmosphere 0.11 (released version)

Every time I go into Animal Crossing New Horizons (1.1.4) and try to enter Edizon this happens.
(Attached screenshot)

Not sure if I'm doing something wrong or missing any files. What could be wrong?

**Comment by jackperry** [comment]
+1. Same thing happening to me on same versions of everything.

**Comment by Waltee** [comment]
Can also confirm that I have this issue, even after updating to the new Tesla menu update that was released 2 hours ago.

**Comment by ppkhoa** [comment]
Same thing happened, even after downgrading to HOS 9.2.0.

To get it to work again, I performed the following:
Rolled back/downgrade to HOS 9.2.0
Removed the new version of nx-ovlloader
Use v1.0.3 of nx-ovlloader, keeping the latest version of Tesla-Menu (ovlloader).
EdiZon via Tesla works again.

Something's wrong with the latest version of nx-ovlloader.

**Comment by Waltee** [comment]
> Something's wrong with the latest version of nx-ovlloader.

I use CFW online so kinda need to have 10.0 which makes this quite a pain that I can't workaround with what you said, unfortunately.

**Comment by Waltee** [comment]
Figured out the issue for me personally. Having "too many" cheats causes it to hard crash. Even though all the cheats worked in the previous version with no issues, I kept getting this crashing issue until the deleted a bunch of the cheats out of my .txt file.

**Comment by ReZeroEmiria** [comment]
> Figured out the issue for me personally. Having "too many" cheats causes it to hard crash. Even though all the cheats worked in the previous version with no issues, I kept getting this crashing issue until the deleted a bunch of the cheats out of my .txt file.

Got an idea of how many is too many? File size or amount you had before vs after?

**Comment by TechGeekGamer** [comment]
Check your cheats here. Don't let it go past the values at the bottom right. https://edizon.werwolv.net/cheat_emu/

**Comment by ppkhoa** [comment]
> 
> 
> Check your cheats here. Don't let it go past the values at the bottom right. https://edizon.werwolv.net/cheat_emu/

This is the solution, at least on HOS 9.2.0, have not tested on HOS 10.0.0 yet. My cheat file contained more than 1024 Total Opcode count, removing the cheats to bring that number below 1024 worked.

**Comment by ReZeroEmiria** [comment]
> > Check your cheats here. Don't let it go past the values at the bottom right. https://edizon.werwolv.net/cheat_emu/
> 
> This is the solution, at least on HOS 9.2.0, have not tested on HOS 10.0.0 yet. My cheat file contained more than 1024 Total Opcode count, removing the cheats to bring that number below 1024 worked.
> 
> Update: Just tested on HOS 10.0.0, worked just fine.

Not sure what the limits are I have a file with higher than 1024 opcode working. Trimming my old file did work though.  Maybe the Opcode limit here is lower than the actual?

**Comment by TechGeekGamer** [comment]
@ReZeroEmiria Just a suggestion; maybe change your issue title to `Launching EdiZon overlay crashes Atmosphere` ?

**Comment by OldGnashburg** [comment]
I'm having the same issue, Edizon Overlay crashes as soon as I press A to select, however it doesn't crash the whole system, just kicks me out of the game and says An Error has Occurred, opcode count isn't an issue.

**Comment by ppkhoa** [comment]
In my case, Atmosphère automatically turn on *all* cheats, which in most cases will cause conflicts and crash the software/game.

If you are using latest Atmosphère, then create atmosphere\loader.ini with the following content (if you want to activate cheat or original album app, then hold L while launching):

[default_config]
override_key=L
cheat_enable_key=L

Also, create atmosphere\system_settings.ini and include the following (just to make sure Atmosphère behave as defined):

; Controls whether dmnt cheats should be toggled on or off by
; default. 1 = toggled on by default, 0 = toggled off by default.
dmnt_cheats_enabled_by_default = u8!0x0
; Controls whether dmnt should always save cheat toggle state
; for restoration on new game launch. 1 = always save toggles,
; 0 = only save toggles if toggle file exists.
dmnt_always_save_cheat_toggles = u8!0x0

I spent 7 hours troubleshooting the vague error message...

**Comment by TechGeekGamer** [comment]
`system-settings.ini` and `loader.ini` should go in `/atmosphere/config/`. Not just `/atmosphere`.

**Comment by MRJ-007** [comment]
OMG I figured out the problem. I deleted the nro file folder for Edizon in the Switch folder. I used Edizon Overlay and cheats work 100% again.

**Comment by TechGeekGamer** [comment]
Yeah, it is a known issue that 3.1.0 version of EdiZon doesn't support Animal Crossing. WerWolv should have it working in 4.0.0.

**Comment by TechGeekGamer** [comment]
@MRJ-007 For the cheats that only some worked, check here. https://edizon.werwolv.net/cheat_emu/

**Comment by Baohdas** [comment]
I have got ATM 0.12 fw 9.2.0 and hakate 5.1.4 when i try to use EEdzone last version atms cresh. I have try the code of @ppkhoa but it chresh. It chresh also if you copy the cheeat fodler in contents folder, pls help
Maybe it can be a solution downgrade atmosphere?

**Comment by TechGeekGamer** [comment]
@Baohdas Try using snapshot version of EdiZon

**Comment by Baohdas** [comment]
Yes I use this one and fail to cresh

**Comment by TechGeekGamer** [comment]
What are you trying to use cheats for?

**Comment by Baohdas** [comment]
no no indipeendeet of the cheat if I start EDizon it creesh, also only to see the build of the game dor example

**Comment by TechGeekGamer** [comment]
Are you using an exFAT formatted SD card

**Comment by Baohdas** [comment]
FAT32

**Comment by TechGeekGamer** [comment]
Do you have  game running in the background?

**Comment by Baohdas** [comment]
I ahve try with running and not

To downgrade atms I can only delete fodelrs and replace them, right?

**Comment by Baohdas** [comment]
OK Guys, With ATMS 0.10.5 The sisteem doesn't do the error and the Switch don't crash. I have not test the overlay and tesla sorry.

**Comment by diaslag13** [comment]
I got the same error. I'm at atmosphere 0.12, HOS 10.0.2, nx-ovlloader 1.5.0, Tesla menu 1.1.3. I tried editing the cheat file and reduce it to only 1 cheat option but it would still crash.

**Comment by mickeyandkaka** [comment]
Same here， looing for solution

**Comment by OldGnashburg** [comment]
So I fixed this issue, it seems to be a memory issue involving sysmodules. However it's really weird, I installed Kosmos with the same number of active sysmodules and it worked fine, then I uninstalled Kosmos and everything still worked fine.
